src/macaron/slsa_analyzer/ci_service/github_actions/analyzer.py

```diff
@@ -0,0 +1,718 @@
+            if not is_expression(java_version):
+                self._lang_versions = [java_version]
+            else:
+                lang_version_exp = java_version
+        # Handle matrix values.
+        matrix_values = {}
+        if lang_distribution_exp and "matrix." in lang_distribution_exp:
+            matrix_values["lang_distribution_var"] = find_expression_variables(
+                value=lang_distribution_exp, exp_var="matrix"
+            )
+        if lang_version_exp and "matrix." in lang_version_exp:
+            matrix_values["lang_version_var"] = find_expression_variables(value=lang_version_exp, exp_var="matrix")
+
+        if matrix_values:
+            job_node = external_node.caller
+            if job_node is None:
+                logger.debug("Unable to find the caller GitHub Action job for step %s.", external_node.name)
+                return
+            try:
+                if (variables := matrix_values.get("lang_distribution_var")) is not None:
+                    values: list[str] = []
+                    for var in variables:
+                        values.extend(resolve_matrix_variable(job_node, var))
+                    if values:
+                        self._lang_distributions = values
+            except GitHubActionsValueError as error:
+                logger.debug(error)
+
+            try:
+                if (variables := matrix_values.get("lang_version_var")) is not None:
+                    values = []
+                    for var in variables:
+                        values.extend(resolve_matrix_variable(job_node, var))
+                    if values:
+                        self._lang_versions = values
+            except GitHubActionsValueError as error:
+                logger.debug(error)
+
+    @property
+    def lang_name(self) -> str:
+        """Get the name of the language."""
+        return self._lang_name
+
+    @property
+    def lang_versions(self) -> list[str] | None:
+        """Get the possible version of the language."""
+        return self._lang_versions
+
+    @property
+    def lang_distributions(self) -> list[str] | None:
+        """Get the possible distributions of the language."""
+        return self._lang_distributions
+
+    @property
+    def lang_url(self) -> str | None:
+        """Get the URL that provides information about the language distributions and versions."""
+        return self._lang_url
+
+
+def create_third_party_action_model(external_node: GitHubWorkflowNode) -> ThirdPartyAction:
+    """Create an instances of third-party model object.
+
+    Parameters
+    ----------
+    external_node: GitHubWorkflowNode
+        The external GitHub Actions workflow node.
+
+    Returns
+    -------
+    ThirdPartyAction
+        An instance object for the ThirdPartyAction model.
+    """
+    action_name = external_node.name
+    action_version = None
+    if "@" in external_node.name:
+        action_name, action_version = external_node.name.split("@")
```

Comment:
Do we assume that `external_node_name` will always be either of these situations:
- Have no "@" symbol
- Have at most one "@" symbol
The reason I asked this question is because `GitHubWorkflowNode.name` value came from the value defined in the workflow:
- https://github.com/oracle/macaron/blob/ed96daea0f0c4ae81f6f42b464d859f0829563aa/src/macaron/slsa_analyzer/ci_service/github_actions/analyzer.py#L293-L299
- https://github.com/oracle/macaron/blob/ed96daea0f0c4ae81f6f42b464d859f0829563aa/src/macaron/slsa_analyzer/ci_service/github_actions/analyzer.py#L248-L25
:thinking:

Reply:
The version can be a tag which theoretically could contain more "@" symbols which would cause this code to error. Add `maxsplit=1` to the `split` call and that will make it behave correctly in all cases.